### ISSUE TITLE
Nav/followbot overwrite 

### ISSUE THREAD

**Issue by SylveonBottle** [nullworks/cathook#718]
Allow the user to take manual control during nav/followbot regardless of nav/followbot settings.

If the user is attempting to use WASD, then we should let the user's own movements overwrite the bot's.

This is so the user can make micro-adjustment of the bot's path, walk out of danger, or temporarily take control to do a minor action (such as upgrading a building or giving a friend a sandvich) without needing to run a command to disable nav/followbot entirely.

**Example types**
**0:** Disabled. Nav/followbot overwrites/ignores all user inputs for nav/followbot's own route.
^Default

**1:** Temporarily disable nav/followbot's movements while user is giving it's own actions.
Used for minor/micro-adjustments of the bot to get it unstuck, but should immediately go back on the path where it last left off.

**2:** Disable navbot/followbot entirely upon receiving an input from the user, then re-enable after X seconds.
This would be used to completely reset the navbot incase it's in an area the user doesn't want - said user could simply press W, begin walking towards a seperate part of the map, then let go and allow the navbot to begin navigating from there.

**3:** Enable navbot upon spawning into the game, disable upon receiving user input. Re-enable after the user respawns.
This would be great for letting the bot walk to the frontlines for the user automatically, and then letting the user play as normal. This would allow for minor automation for navbots to be used by all users as a small automation tool for getting into the front-lines.

**TL;DR:** If the user is spamming WASD then chances are they're panicking - and if they're panicking, then chances are we should probably assume the navbot is about to do something stupid and let the user control it temporarily.

**Comment by SylveonBottle** [comment]
Btw 3 is my fav and could be it's own thing, like `nav-bot.walk-out-of-spawn` or something lol. It's just a minor thing but it would be cool if I die and go and piss, the bot can start playing automatically for me and when I get back I just take control.

**Comment by EricSmith17** [comment]
Sorry, but what the fuck youre talking about ? 
Just break navbot if movement keys are pressed and then continue.
Navbot is choosing where to walk by situations.

bool useNavBot = true;
if (keys_pressed("W", "A", "S", "D" */ etc /*) == true && useNavBot == true)
{
         useNavBot = false;
} 
elseif (player_spawn(pLocal))
{
         useNavBot = true;
}

**Comment by TotallyNotElite** [comment]
> 
> 
> Sorry, but what the fuck youre talking about ?
> Just break navbot if movement keys are pressed and then continue.
> Navbot is choosing where to walk by situations.
> 
> ```
> bool useNavBot = true;
> if (keys_pressed("W", "A", "S", "D" */ etc /*) == true && useNavBot == true)
> {
>          useNavBot = false;
> } 
> elseif (player_spawn(pLocal))
> {
>          useNavBot = true;
> }
> ```

First of all, please watch your language. Second of all, this solution is not ideal at all because it's sometimes advantageous to stay inside spawn instead of blindly running out (note: pressing a key would instantly restart navbot again). That could easily be solved by checking for the player spawned event and/or adding a button for initiating navbot.

**Comment by BenCat07** [comment]
+ keys are not reliable. I have seen people with arrow keys bound to movement
